File: competitive_programming/codeforces/11_golang/339a_helpful_maths.go
package main

import (
	"fmt"
	"strings"
	"sort"
)

func helpfulMaths(expression string) string {
	parts := strings.Split(expression, "+")
	sort.Strings(parts)
	return strings.Join(parts, "+")
}

func main() {
	if helpfulMaths("3+2+1") != "1+2+3" || helpfulMaths("1+1+3+1+3") != "1+1+1+3+3" {
		panic("helpful maths failed")
	}
	fmt.Println("339A helpful maths ok")
}

File: competitive_programming/codeforces/07_typescript/339a_helpful_maths.ts
function helpfulMaths(expression: string): string {
  return expression.split("+").sort().join("+");
}

if (helpfulMaths("3+2+1") !== "1+2+3") throw new Error("helpful maths failed");
if (helpfulMaths("1+1+3+1+3") !== "1+1+1+3+3") throw new Error("helpful maths failed");
console.log("339A helpful maths ok");
